lib/failure.sh

```diff
@@ -978,3 +978,17 @@ warn_multiple_pnpm_version() {
        To ensure we install the version of pnpm you want, remove one of these fields."
   mcount 'warnings.pnpm.multiple-version'
 }
+
+warn_skipping_unsafe_pnpm_prune() {
+  local pnpm_version="$1"
+  warn "Pruning skipped due to presence of pre/post install scripts
+
+       The version of pnpm used ($pnpm_version) will execute any 'preinstall' or 'postinstall' scripts
+       declared in package.json during pruning which can cause build failures. Since pruning can't
+       be done safely for your build, it will be skipped.
+
+       To fix this you can either:
+       - Upgrade your version of pnpm to 8.15.6 or higher.
+       - If possible, remove any 'preinstall' or 'postinstall' scripts from package.json."
```

Comment:
I think I'd favor dropping this suggestion. Most folks with a `preinstall` or `postinstall` probably have it for a good reason. `preinstall` and `postinstall` are canonical and ubiquitous - moving/removing these scripts possibly affects other workflows (aside from deploying to heroku) and probably makes life harder. 

Upgrading to `pnpm@8.15.6` is a much more reasonable ask. With `pnpm` support being brand new to this buildpack there also isn't a need to support a long backlog of pnpm versions.

Reply:
Yeah, and now that the list includes more than just the `preinstall` and `postinstall` scripts this suggestion gets even more awkward. Changed the warning to simply note the scripts that pruning can't be done safely with for `pnpm<8.15.6` and recommend the user upgrades.